### ISSUE TITLE
Misleading: warning csolution: key 'selected-by' was not recognized

### ISSUE THREAD

**Issue by ReinhardKeil** [Open-CMSIS-Pack/devtools#1402]
**Describe the bug**
The key `selected-by` is renamed to `selected-by-pack` the tool creates a warning that is confusing.

```
C:\Test\Test2\TestM85>cbuild testm85.csolution.yml
info cbuild: Build Invocation 2.3.0-dev0 (C) 2024 Arm Ltd. and Contributors
C:/Test/Test2/TestM85/TestM85.cbuild-pack.yml:4:7 - warning csolution: key 'selected-by' was not recognized
C:/Test/Test2/TestM85/TestM85.cbuild-pack.yml:7:7 - warning csolution: key 'selected-by' was not recognized
```

**To Reproduce**
Use a project with `*.cbuild-pack.yml` that was previously translated with Toolbox 2.2.0

**Expected behavior**
The fact that the key is renamed is irrelevant and should be tolerated.

**Environment:**

- CMSIS-Toolbox 2.3.0-dev0

**Comment by brondani** [comment]
It turns out the `selected-by` key mentioned in the warning does not come from the tool but from the *.cbuild-pack.yml file created by an older version.
For backward compatibility the tool must cope with both `selected-by` and `selected-by-pack`.

**Comment by jkrech** [comment]
Fixed in CMSIS-Toolbox 2.3.0
